src/server.rs

```diff
@@ -1102,6 +1102,105 @@ impl Default for Builder {
 // ===== impl SendResponse =====
 
 impl<B: Buf> SendResponse<B> {
+    /// Send an interim informational response (1xx status codes)
+    ///
+    /// This method can be called multiple times before calling `send_response()`
+    /// to send the final response. Only 1xx status codes are allowed.
+    ///
+    /// Interim informational responses are used to provide early feedback to the client
+    /// before the final response is ready. Common examples include:
+    /// - 100 Continue: Indicates the client should continue with the request
+    /// - 103 Early Hints: Provides early hints about resources to preload
+    ///
+    /// # Arguments
+    /// * `response` - HTTP response with 1xx status code and headers
+    ///
+    /// # Returns
+    /// * `Ok(())` - Interim Informational response sent successfully
+    /// * `Err(Error)` - Failed to send (invalid status code, connection error, etc.)
+    ///
+    /// # Examples
+    /// ```rust
+    /// use h2::server;
+    /// use http::{Response, StatusCode};
+    ///
+    /// # async fn example(mut send_response: h2::server::SendResponse<bytes::Bytes>) -> Result<(), h2::Error> {
+    /// // Send 100 Continue before processing request body
+    /// let continue_response = Response::builder()
+    ///     .status(StatusCode::CONTINUE)
+    ///     .body(())
+    ///     .unwrap();
+    /// send_response.send_informational(continue_response)?;
+    ///
+    /// // Later send the final response
+    /// let final_response = Response::builder()
+    ///     .status(StatusCode::OK)
+    ///     .body(())
+    ///     .unwrap();
+    /// let _stream = send_response.send_response(final_response, false)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// # Errors
+    /// This method will return an error if:
+    /// - The response status code is not in the 1xx range
+    /// - The final response has already been sent
+    /// - There is a connection-level error
+    pub fn send_informational(&mut self, response: Response<()>) -> Result<(), crate::Error> {
+        let stream_id = self.inner.stream_id();
+        let status = response.status();
+
+        tracing::trace!(
+            "send_informational called with status: {} on stream: {:?}",
+            status,
+            stream_id
+        );
+
+        // Validate that this is an informational response (1xx status code)
+        if !response.status().is_informational() {
```

Comment:
I noticed that these same checks also happen in `Send::send_interim_informational_headers()`, and in `Streams::send_informational_headers()`, which is called lower in this method, but there it will also emit helpful traces.

Is this the "only" place that could call the internal methods? If so, perhaps here is where the checks, traces, and error messages should happen. And later on, it can just be `assert!(the_things)` to catch if any other code path is added that doesn't check. Because at that point, the contract is on "us", the library, to uphold the invariants.

What do you think?